Add the ability to add keys to the union after it has been created

Marwes opened the pull request:
For typechecking in https://github.com/Marwes/embed_lang I do not know how many variables I need beforehand so I added the ability to add new keys after the Union was created. Figure it might be useful upstream as well. If you have any ideas for how to make this better I am happy to update this PR.

gifnksm commented:
Thank you for your pull request! :smile_cat:

I agree that adding new keys is very useful in many case. I think this library should provide such a feature.

Would you squash the commits into one commit after addressing my comments?

Marwes commented:
Rebased with comments in mind. I renamed it to `insert` as `push` seems to imply some order of the inserted keys which is really an implementation detail.

gifnksm commented:
Merged. Thank you!

Marwes commented:
Any chance you could push this to crates.io so I can use this directly instead of a git dependency on my own repo? :)

gifnksm commented:
Pushed v0.2.2 to crates.io. Sorry for being late.

Marwes commented:
Nice, thank you!
